pywr/parameters/control_curves.py

```diff
@@ -3,79 +3,142 @@
 """
 
 from ._control_curves import BaseControlCurveParameter, ControlCurveInterpolatedParameter
-from .parameters import parameter_registry
+from .parameters import parameter_registry, load_parameter_values, load_parameter
 import numpy as np
 
 
-class ControlCurvePiecewiseParameter(BaseControlCurveParameter):
-    """ A control curve Parameter that returns one of two values depending on whether the current
-    volume is above or below the control curve.
+class ControlCurveParameter(BaseControlCurveParameter):
+    """ A generic multi-levelled control curve Parameter.
+
+     This parameter can be used to return different values when a `Storage` node's current
+      volumes is at different percentage of `max_volume` relative to predefined control curves.
+      Control curves must be defined in the range [0, 1] corresponding to 0% and 100% volume.
+
+     By default this parameter returns an integer sequence from zero if the first control curve
+      is passed, and incrementing by one for each control curve (or "level") the `Storage` node
+      is below.
+
+    Parameters
```

Comment:
This belongs in the `__init__` docstring. The docstring for the class should include "Attributes" (with some duplication of names in this case) and "Methods" (which I don't think we need atm). https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#documenting-classes